test/t8_gtest_nca.cxx

```diff
@@ -0,0 +1,150 @@
+    const int num_children = ts->t8_element_num_children (correct_nca);
+    /* Iterate over all combinations of two children from correct_nca*/
+    for (i = 0; i < num_children; i++) {
+        ts->t8_element_child (correct_nca, i, desc_a);
+        for (j = i + 1; j < num_children; j++) {
+            ts->t8_element_child (correct_nca, j, desc_b);
+            /*Compute the nca */
+            ts->t8_element_nca (desc_a, desc_b, check);
+            /*expect equality */
+            EXPECT_TRUE ((ts->t8_element_compare (correct_nca, check) == 0));
+        }
+    }
+
+}
+
+/**
+ * Check the nca for elements on the maximal level.
+ * We iteratively compute an element that is the correct nca up to level 10.
+ * The first and the last descendant of this element at the maximal level are
+ */
+TEST_P(nca, nca_check_deep){
+    const int       max_lvl = 10;
+    const int       elem_max_level = ts->t8_element_maxlevel();
+    /* num_children is not a const here, cause this can change for pyramids*/
+    int             num_children;
+    /* iterate over levels and children*/
+    int             lvl, check_lvl, child_id;
+    t8_element_t    *tmp;
+
+    ts->t8_element_new(1, &tmp);
+    ts->t8_element_copy(correct_nca, tmp);
+    for(lvl = 1; lvl <= max_lvl; lvl++){
+        num_children = ts->t8_element_num_children(correct_nca);
+        for(child_id = 0; child_id < num_children; child_id++){
+            ts->t8_element_child(tmp, child_id, correct_nca);
+            /* Compute first and last descendant at every level up to elem_max_lvl. 
+                *They have the correct_nca as the nca*/
+            for(check_lvl = lvl + 1; check_lvl < elem_max_level; check_lvl++){
+                ts->t8_element_first_descendant(correct_nca, desc_a, check_lvl);
+                ts->t8_element_last_descendant(correct_nca, desc_b, check_lvl);
+                /* Compute the nca of desc_a and desc_b*/
+                ts->t8_element_nca(desc_a, desc_b, check);
+                if(eclass == T8_ECLASS_VERTEX){
+                    /* first- last-descendant logic does not hold for vertices.*/
+                    EXPECT_EQ(ts->t8_element_level(check), SC_MIN(ts->t8_element_level(desc_a), ts->t8_element_level(desc_b)));
+                }
+                else{
+                    /* Expect equality of correct_nca and check for every other class*/
+                    EXPECT_TRUE ((ts->t8_element_compare (correct_nca, check) == 0));
+                }
+            }
+        }
+        /*Determine next element*/
+        if(num_children != 1){
+            ts->t8_element_copy(tmp, correct_nca);
+            /*Continue in the middle*/
+            ts->t8_element_child(correct_nca, num_children/2, tmp);
+        }
+        else{
+            ts->t8_element_copy(correct_nca, tmp);
+        }
+    }
+    ts->t8_element_destroy(1, &tmp);
+}
+
+
+INSTANTIATE_TEST_SUITE_P (t8_gtest_nca, nca,
+                        testing::Range (T8_ECLASS_ZERO, T8_ECLASS_COUNT));
```

Comment:
You could also exclude Pyramids by adjusting the range here, but I think your solution is fine.
```suggestion
                        testing::Range (T8_ECLASS_ZERO, T8_ECLASS_PYRAMID));
```

Reply:
Yes, I've done that first, but with the current setup you see that test is skipped for pyramids without having a detailed look into the file.